graderx/graders/stdout_graders/c/c_grader.py

```diff
@@ -123,23 +123,23 @@ def get_diff_results_file(course_name, lab):
         data = json.load(f)
     return data
 
-def get_submission_files(course, lab, submission_id):
+def get_submission_files(course, lab, submission_id, submission_key):
     lab_path = get_lab_path(course, lab)
-    return list(lab_path.glob(f"submissions/{submission_id}/*.c"))
+    return list(lab_path.glob(f"submissions/{submission_key}/{submission_id}/*.c"))
```

Comment:
same note for manually joining paths. Maybe you'd like to have a `PathManager` to group all the joins in one place

Reply:
What I understood with PathManager is a single module that has all the needed paths but with descriptive names, did I get it right?

Reply:
yes, or *composes* the paths so it'll have something like `def glob_submission_files(submission_key, submission_id)`

I mean, it doesn't need to have just constant fields